core/src/main/java/org/elasticsearch/action/percolate/PercolateResponse.java

```diff
@@ -136,10 +128,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                     builder.field(Fields._INDEX, match.getIndex());
                     builder.field(Fields._ID, match.getId());
                     float score = match.getScore();
-                    if (score != PercolatorService.NO_SCORE) {
+                    if (score != NO_SCORE) {
                         builder.field(Fields._SCORE, match.getScore());
                     }
-                    if (match.getHighlightFields() != null) {
+                    if (match.getHighlightFields() != null && match.getHighlightFields().isEmpty() == false) {
```

Comment:
Can we make getHighlightFields always return a non-null value? (using Collections.emytyXXX if necessary)